src/maidsafe/vault/mpid_manager/messages.cc

```diff
@@ -0,0 +1,138 @@
+/*  Copyright 2015 MaidSafe.net limited
+
+    This MaidSafe Software is licensed to you under (1) the MaidSafe.net Commercial License,
+    version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+    licence you accepted on initial access to the Software (the "Licences").
+
+    By contributing code to the MaidSafe Software, or to this project generally, you agree to be
+    bound by the terms of the MaidSafe Contributor Agreement, version 1.0, found in the root
+    directory of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also
+    available at: http://www.maidsafe.net/licenses
+
+    Unless required by applicable law or agreed to in writing, the MaidSafe Software distributed
+    under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+    OF ANY KIND, either express or implied.
+
+    See the Licences for the specific language governing permissions and limitations relating to
+    use of the MaidSafe Software.                                                                 */
+
+#include "maidsafe/vault/mpid_manager/messages.h"
+
+namespace maidsafe {
+
+namespace vault {
+
+// ================================= MpidMessageBase =============================================
+
+MpidMessageBase::MpidMessageBase()
+    : sender(), receiver(), id(), parent_id(), signed_header() {}
+
+MpidMessageBase::MpidMessageBase(const std::string& serialised_copy)
+    : sender(), receiver(), id(), parent_id(), signed_header() {
+  InputVectorStream binary_input_stream {
+      SerialisedData(serialised_copy.begin(), serialised_copy.end()) };
+  Parse(binary_input_stream, sender, receiver, id, parent_id, signed_header);
+}
+
+MpidMessageBase::MpidMessageBase(const Identity& sender_in,
+                                 const Identity& receiver_in,
+                                 int32_t id_in,
+                                 int32_t parent_id_in,
+                                 const MessageHeaderType& signed_header_in)
+    : sender(sender_in), receiver(receiver_in), id(id_in), parent_id(parent_id_in),
+      signed_header(signed_header_in) {}
+
+MpidMessageBase::MpidMessageBase(const MpidMessageBase& other)
+    : sender(other.sender), receiver(other.receiver), id(other.id), parent_id(other.parent_id),
+      signed_header(other.signed_header) {}
+
+MpidMessageBase::MpidMessageBase(MpidMessageBase&& other)
+    : sender(std::move(other.sender)), receiver(std::move(other.receiver)), id(std::move(other.id)),
+      parent_id(std::move(other.parent_id)), signed_header(std::move(other.signed_header)) {}
+
+MpidMessageBase& MpidMessageBase::operator=(MpidMessageBase other) {
```

Comment:
We don't use the copy-and-swap idiom any more.  The copy c'tor and copy assignment operator can be written ` = default;`, but we need to hand-write the move ones for MSVC.